worker/worker.go

```diff
@@ -87,7 +91,14 @@ func (w *Worker) run() {
 			return
 		case <-ticker.C:
 			for {
-				ctx := context.TODO()
+				// TODO: start trace with task context
+				ctx := context.Background()
+				if w.tracer != nil {
+					start := time.Now()
+					tracingCtx, span := w.tracer.Start(ctx, "worker.fetch", trace.WithSpanKind(trace.SpanKindConsumer), trace.WithTimestamp(start))
+					defer span.End()
```

Comment:
will this cause a memory leak when the for loop never exists? (The for loop only exists when the queue is empty)